### PR TITLE
Migrate mermaid diagrams to markdoc

### PR DESCRIPTION
**Pull request by tom-blake**
Migrate all MDX files containing mermaid diagrams to markdoc. It was easier to migrate all MDX files than it was to support mermaid with both MDX and markdoc.

Additional necessary changes include:
- Create a custom image component for markdoc. This also meant moving the farmlands card image out of the public folder
- Removing our custom mermaid plugin in favour of mermaid's out of the box configuration.
- Updating the markdoc integration config to ignore identation to support better readability.

Test plan:
- Confirm no visual regressions in each migrated guide using the dev environment

### CODE REVIEW

File: src/components/Figure.astro [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+// https://docs.astro.build/en/guides/integrations-guide/markdoc/#custom-image-components
+// https://discord.com/channels/830184174198718474/1182681722215743578/1204466454922002452
+// Open Github issue: https://github.com/withastro/astro/pull/11761
```
Comment (tom-blake) [inline src/components/Figure.astro:4]: This Github issue will be fixed in the next Github release so we won't need to hack around this

File: markdoc.config.mjs [hunk 2]
```diff
@@ -12,11 +12,43 @@ export default defineMarkdocConfig({
...
+        __optimizedSrc: { type: 'Object' },
+      },
     },
     link: {
       ...nodes.link,
       render: component('./src/components/Link.astro'),
     },
   },
+  tags: {
+    farmlandsSolutionCard: {
```
Comment (MeganSteenkamp) [inline markdoc.config.mjs:34]: Suggestion: This component could be something more generic. This is probably outside of the scope of your change so can leave it 🤷‍♀️

File: markdoc.config.mjs [hunk 3]
```diff
@@ -12,11 +12,43 @@ export default defineMarkdocConfig({
...
+      render: component('./src/components/SolutionCard.astro'),
+      attributes: {
+        description: { type: String },
+        imageSrc: { type: String },
+        linkText: { type: String },
+        linkUrl: { type: String },
+        title: { type: String },
+      },
+    },
+    gridList: {
```
Comment (MeganSteenkamp) [inline markdoc.config.mjs:44]: Suggestion: This is just an empty grid with no styling 🤨 
Perhaps it should be a container if it's something we will reuse?

File: src/components/Mermaid.astro [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+const {code} = Astro.props;
+---
+<pre class="mermaid my-8 rounded-lg border border-outline-opaque bg-surface-secondary">
```
Comment (MeganSteenkamp) [inline src/components/Mermaid.astro:4]: Change: Center the mermaid diagrams. 
You can do this by wrapping the `pre` element with a `center` element.

Before:
image

After:
image

File: src/content/guides/farmlands-pos-integration.mdoc [hunk 5]
```diff
@@ -61,12 +59,11 @@ sequenceDiagram
 
     > The Card number is under the Cardholder’s name on the front of the card; and under the barcode on the back of the card.
 
-<img src="/farmlands-card.png" alt="The back of an example Farmlands card, displaying a barcode and a 9-digit card number" style="height: 288px;"/>
+    ![The back of an example Farmlands card, displaying a barcode and a 9-digit card number](../../assets/images/farmlands-card.png)
```
Comment (MeganSteenkamp) [inline src/content/guides/farmlands-pos-integration.mdoc:62]: @tom-blake Suggestion: Is there a way to set a max width? Or should we make the image itself smaller?
It's looking a little large 😅 

image

File: src/content/guides/merchant-integration-qr-code-flow.mdoc [hunk 6]
```diff
@@ -40,10 +38,9 @@ sequenceDiagram
 
 1. The POS [creates a Payment Request](/api/payment-requests#create-a-payment-request) and presents a QR code to the Patron on a customer-facing display.
 
-  > The QR code decodes to a URL of the form `https://service.centrapay.com/api/payment-requests/{paymentRequestId}/pay`
+    > The QR code decodes to a URL of the form `https://service.centrapay.com/api/payment-requests/{paymentRequestId}/pay`
```
Comment (MeganSteenkamp) [inline src/content/guides/merchant-integration-qr-code-flow.mdoc:41]: @tom-blake Question: I know you're refactoring but is this correct? 🤨 
It normally decodes to `https://app.centrapay.com/pay/{paymentRequestId}` - is it different for POS integrations?

**Review by MeganSteenkamp — APPROVED** [review]

**Pull request merged**